src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -205,6 +210,14 @@ static Object preProcessValue(Object value, Schema schema, Schema newSchema) {
         ArrayList<Struct> mapStructs = new ArrayList<>();
         Map<?, ?> map = (Map<?, ?>) value;
         Schema newValueSchema = newSchema.valueSchema();
```

Comment:
Move below, or better reuse instead of `newSchema.valueSchema()` in the "string" case.